pkg/apiserver/webhook/configmap_webhook.go

```diff
@@ -0,0 +1,130 @@
+		return v.validateCreate()
+	case admissionv1.Update:
+		return v.validateUpdate(req)
+	case admissionv1.Delete:
+		return v.validateDelete()
+	default:
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf("invalid admission webhook request"))
+	}
+}
+
+// InjectDecoder injects the decoder.
+func (v *ConfigMapValidator) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
+
+// getCloudResourcePrefix get the CloudResourcePrefix from the ConfigMap.
+func (v *ConfigMapValidator) getCloudResourcePrefix(configMap *corev1.ConfigMap) (string, error) {
+	configData := configMap.Data["nephe-controller.conf"]
+	controllerConfig := &config.ControllerConfig{}
+	err := yaml.UnmarshalStrict([]byte(configData), controllerConfig)
+	if err != nil {
+		return "", err
+	}
+	return controllerConfig.CloudResourcePrefix, nil
+}
+
+// validateCreate validates the ConfigMap create.
+func (v *ConfigMapValidator) validateCreate() admission.Response {
+	return admission.Allowed("")
+}
+
+// validateUpdate validates the ConfigMap and CloudResourcePrefix update.
+func (v *ConfigMapValidator) validateUpdate(req admission.Request) admission.Response {
+	newConfigMap := &corev1.ConfigMap{}
+	err := v.decoder.Decode(req, newConfigMap)
+	if err != nil {
+		v.Log.Error(err, "failed to decode configMap", "ConfigMapValidator", req.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if newConfigMap.GetName() != config.ConfigMap {
+		return admission.Allowed("")
+	}
+
+	newCloudResourcePrefix, err := v.getCloudResourcePrefix(newConfigMap)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	oldConfigMap := &corev1.ConfigMap{}
+	err = json.Unmarshal(req.OldObject.Raw, &oldConfigMap)
+	if err != nil {
+		v.Log.Error(err, "failed to decode old configMap", "ConfigMapValidator", req.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	oldCloudResourcePrefix, err := v.getCloudResourcePrefix(oldConfigMap)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	if newCloudResourcePrefix == oldCloudResourcePrefix {
+		v.Log.V(1).Info("CloudResourcePrefix are same", "newCloudResourcePrefix",
+			newCloudResourcePrefix, "oldCloudResourcePrefix", oldCloudResourcePrefix)
+		return admission.Allowed("")
+	}
+	if v.NpController.IsCloudResourceCreated() {
+		return admission.Denied(fmt.Sprintf("cloud resource is already created with CloudResourcePrefix %s. "+
+			"Please delete the resource and try or use the same CloudResourcePrefix",
+			oldCloudResourcePrefix))
+	}
+	if len(newCloudResourcePrefix) > 0 {
+		if !config.ValidateName(newCloudResourcePrefix) {
+			return admission.Denied(fmt.Sprintf("invalid CloudResourcePrefix %s only "+
+				"alphanumeric and '-' characters are allowed.", newCloudResourcePrefix))
+		}
+	}
+	return admission.Allowed("")
+}
+
+// validateDelete validates the ConfigMap delete.
+func (v *ConfigMapValidator) validateDelete() admission.Response {
```

Comment:
why no check for delete case.. lets say, I used prefix as xyz and then cloud resources are created, later, I delete prefix xyz, you will fail back to default right? what would happen to resources with xyz?

Reply:
If we delete the config map the pod will be crashed